### ISSUE TITLE
YouTube won't login even if YouTube and .google.com are both in the same container

### ISSUE THREAD

**Issue by abitawoods1** [mozilla/multi-account-containers#2532]
### Before submitting a bug report

- [X] I updated to the latest version of Multi-Account Container and tested if I can reproduce the issue
- [X] I searched for existing reports to see if it hasn't already been reported

### Step to reproduce

1.  created youTube container
2. go to YouTube, which opens in that container, and attempt to open a movie which requires a google login
3a.  opens separate tab with google login (no container
3b.  login, there.
3c.  the google tab closes (and logs me in)
3d.  YouTube tab throw a new tab (in the youTube container which has the error (...there s a problem .... )
3d.  i.e., login fails
4. go back and separate execute a google login, and assign the domain (accounts.google.com) to youTube container
5. close all tabs (this clears all cookies on my install)
6. go to youTube and attempt movie again; google login opens in same tab, logs in, and new tab pops up with the same youTube error.   google login, on the original tab, hangs

result:  login has same failure anyway

### Actual behavior

.

### Expected behavior

Should login

Should not require me to set the 'limit other domains' switch, on the youTube container.
I would like to be protected if i click on a URL somewhere in a YouTube tab, which opens another domain, and have that domain open elsewhere, without a container, or with one if i ve set it so.

### Provide a copy of Troubleshooting Information page (optional)

```plain text
prior opened bug suggested two steps
1. add google to domains in the YouTube container
2. turn off the limit switch.

that bug was closed when reporting person was satisfied with #2
i m trying to get method 1 to work as expected

thanks !
```

**Comment by justsomeguyonacomputer** [comment]
I am having this same issue.

I have all google pages in one container, & would like to force all google services to remain in that container & not leak to other sites. I can log into gmail, voice, drive, all of the other services. When I try to log into youtube it opens a new tab that ultimately takes me to an error page stating there was a problem logging me in.

All of the pages that open have been added to this same single container. The only work around is to disable the "Limit to designated sites" switch. That defeats the purpose of the containers.

Is there any update on this bug?

Thank you for your time!

**Comment by creative-resort** [comment]
Did you set for this container "limit to designated sites"?
Remove this setting, to allow Google to hop to e.g. their accounts or any other necessary subdomain, while staying within the container.

**Comment by JohnyP36** [comment]
> Did you set for this container "limit to designated sites"?
Yes, as I don't want to open other e.g. news website in this container. But thanks for the suggestion.

**Comment by creative-resort** [comment]
If you don't need Google and associated domains outside of this container, you can add all necessary subdomains also to this container.
Have a look at this add-on for rule-based assigning, e.g. for wildcards:
https://addons.mozilla.org/en-US/firefox/addon/containerise/

**Comment by firepainting** [comment]
@creative-resort 
> Have a look at this add-on for rule-based assigning, e.g. for wildcards: https://addons.mozilla.org/en-US/firefox/addon/containerise/

I'm trying to switch to FF Multi-Account Containers because Containerise hasn't been updated for almost 5 years now. Containerise is great and underrated.

For this particular Issue (which I'm also experiencing right now), the real cause seems to be the quick **redirects**. 

Correct me if I'm wrong. Specific subdomains can be assigned to FF MACs if those subdomains can fully load first, i.e., they are not redirects.

I find that in logging in on YouTube, we go through 4 different (sub)domains, in this order:

- `www.youtube.com`
- `accounts.google.com` (where we log in)
- `myaccount.google.com` (the quick redirect, this "page" doesn't fully load)
- `www.google.com` (this is actually another redirect, but it now fully loads because the process has stopped, likely due to the previous subdomain `myaccount.google.com` having lost contact) 
- If redirecting was uninterrupted at this point, you should be returned to `www.youtube.com`.

The question here seems to be **why isn't there an input field where users can assign a set of subdomains/URLs to a specific container?** Containerise has that, it's glorious.

I haven't tried logging in on Microsoft/Bing yet, but I'm pretty sure it's the same story. MS uses a bit more redirects than signing in on YouTube.

**Comment by MrPistachios** [comment]
ok so the domains you listed gave me an idea and it might have fixed the issue for me

I went to each of the domains you listed and then clicked the 'always open site in container' button to add them to my google container rules.

the only one that was missing was `accounts.youtube.com` I found it by screen recording the url and then playing it back at slow speed
when I went to that site it was a 404 google page but I was still able to select to always open in the google container, after adding it my youtube account successfully signed in.

my account was already logged in before hand through gmail but would normally be unable to login to youtube unless I removed the checkbox to only allow these sites in that container

**Comment by dmiranda12** [comment]
> I find that in logging in on YouTube, we go through 4 different (sub)domains, in this order:
> 
>     * `www.youtube.com`
> 
>     * `accounts.google.com` (where we log in)
> 
>     * `myaccount.google.com` (the quick redirect, this "page" doesn't fully load)
> 
>     * `www.google.com` (this is actually another redirect, but it now fully loads because the process has stopped, likely due to the 

In my case, adding https://accounts.google.com/  to the google container beforehand did the trick.
